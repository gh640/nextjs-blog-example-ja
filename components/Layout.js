import Head from "next/head"
import Link from "next/link"

const Layout = (props) => {
  const { title, children } = props
  const siteTitle = "後藤のブログ"

  return (
    <div className="page">
      <Head>
        <title>{title ? `${title} | ${siteTitle}` : siteTitle}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <header>
        <h1 className="site-title">
          <Link href="/">
            <a>{siteTitle}</a>
          </Link>
        </h1>
      </header>

      <main>
        {title ? <h1 className="page-title">{title}</h1> : ``}
        <div className="page-main">
          {children}
        </div>
      </main>

      <footer>
        &copy; {siteTitle}
      </footer>

      <style jsx>{`
        .page {
          padding: 2em 1em;
          max-width: 800px;
          margin-left: auto;
          margin-right: auto;
        }

        header {
          display: flex;
          justify-content: center;
          align-items: center;
          margin: 0 0 4em;
        }

        .site-title a {
          color: inherit;
          text-decoration: none;
        }

        footer {
          margin-top: 4em;
          padding-top: 2em;
          padding-bottom: 2em;
          display: flex;
          justify-content: center;
          align-items: center;
        }
      `}</style>

      <style jsx global>{`
        html,
        body {
          padding: 0;
          margin: 0;
          font-family: 'Noto Sans JP', -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
          color: #222;
        }

        img,
        iframe {
          max-width: 100%;
        }

        h1, h2, h3, h4, h5, h6 {
          font-family: Montserrat, -apple-system, "Segoe UI", "Helvetica Neue",
            "Hiragino Kaku Gothic ProN", メイリオ, meiryo, sans-serif;
        }

        * {
          box-sizing: border-box;
        }
      `}</style>
    </div>
  )
}

export default Layout
