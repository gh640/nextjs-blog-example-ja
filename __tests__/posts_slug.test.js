import { cloneElement } from "react"
import renderer from "react-test-renderer"

import Layout from "../components/Layout"
import Post from "../pages/posts/[slug].js"

jest.mock("../components/Layout")

describe("Post", () => {
	Layout.mockImplementation(({ title, children }) => (
		<div>
			<span>{title}</span>
			{children}
		</div>
	))

	it("renders correctly with markdown", () => {
		const content = [
			"## 見出し 2",
			"",
			"昔々あるところに。",
		].join("\n")
		const tree = renderer
			.create(<Post 
				title="はじめての投稿" 
				published="2030/10/20" 
				content={content} 
			/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it("renders correctly with raw html", () => {
		const content = [
			"<h3>テーブルサンプル</h3>",
			"<div>",
			"	<table>",
			"		<thead>",
			"			<tr>",
			"				<th>カラム 1</th>",
			"				<th>カラム 2</th>",
			"			</tr>",
			"		</thead>",
			"		<tbody>",
			"			<tr>",
			"				<td>1.1</td>",
			"				<th>1.2</th>",
			"			</tr>",
			"			<tr>",
			"				<td>2.1</td>",
			"				<th>2.2</th>",
			"			</tr>",
			"		</tbody>",
			"	</table>",
			"</div>",
		].join("\n")
		const tree = renderer
			.create(<Post
				title="はじめての投稿"
				published="2030/10/20"
				content={content}
			/>)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
