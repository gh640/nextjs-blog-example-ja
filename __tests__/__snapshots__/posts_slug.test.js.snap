// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post renders correctly with markdown 1`] = `
<div>
  <span>
    はじめての投稿
  </span>
  <div
    className="post-meta"
  >
    <span>
      2030/10/20
    </span>
  </div>
  <div
    className="post-body"
  >
    <h2>
      見出し 2
    </h2>
    

    <p>
      昔々あるところに。
    </p>
  </div>
</div>
`;

exports[`Post renders correctly with raw html 1`] = `
<div>
  <span>
    はじめての投稿
  </span>
  <div
    className="post-meta"
  >
    <span>
      2030/10/20
    </span>
  </div>
  <div
    className="post-body"
  >
    <h3>
      テーブルサンプル
    </h3>
    

    <div>
      
	
      <table>
        <thead>
          <tr>
            <th>
              カラム 1
            </th>
            <th>
              カラム 2
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1.1
            </td>
            <th>
              1.2
            </th>
          </tr>
          <tr>
            <td>
              2.1
            </td>
            <th>
              2.2
            </th>
          </tr>
        </tbody>
      </table>
      

    </div>
  </div>
</div>
`;
